src/model.ts

```diff
@@ -0,0 +1,189 @@
+ *
+ * @example
+ * const {BigQuery} = require('@google-cloud/bigquery');
+ * const bigquery = new BigQuery();
+ * const dataset = bigquery.dataset('my-dataset');
+ *
+ * const model = dataset.model('my-model');
+ */
+class Model extends common.ServiceObject {
+  constructor(dataset: Dataset, id: string) {
+    const methods = {
+      /**
+       * Delete a model.
+       *
+       * @see [Models: delete API Documentation]{@link https://cloud.google.com/bigquery/docs/reference/rest/v2/models/delete}
+       *
+       * @method Model#delete
+       * @param {function} [callback] The callback function.
+       * @param {?error} callback.err An error returned while making this
+       *     request.
+       * @param {object} callback.apiResponse The full API response.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.delete((err, apiResponse) => {});
+       *
+       * @example <caption>If the callback is omitted we'll return a Promise
+       * </caption>
+       * const [apiResponse] = await model.delete();
+       */
+      delete: true,
+
+      /**
+       * Check if the model exists.
+       *
+       * @method Model#exists
+       * @param {function} [callback] The callback function.
+       * @param {?error} callback.err An error returned while making this
+       *     request.
+       * @param {boolean} callback.exists Whether the model exists or not.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.exists((err, exists) => {});
+       *
+       * @example <caption>If the callback is omitted we'll return a Promise
+       * </caption>
+       * const [exists] = await model.exists();
+       */
+      exists: true,
+
+      /**
+       * Get a model if it exists.
+       *
+       * @see [Models: get API Documentation]{@link https://cloud.google.com/bigquery/docs/reference/rest/v2/models/get}
+       *
+       * @method Model#get:
+       * @param {function} [callback] The callback function.
+       * @param {?error} callback.err An error returned while making this
+       *     request.
+       * @param {Model} callback.model The {@link Model}.
+       * @param {object} callback.apiResponse The full API response.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.get((err, model2, apiResponse) => {
```

Comment:
Can we just show it with the `err` argument, so we don't have to do the `model2` thing? The comment mentions how to get the metadata from the original object, so the natural question people would have is "what is the difference between model and model2? Do I need it?"

Reply:
Yeah, that example felt kinda weird, I just went that way for completeness.